core/templates/services/svg-sanitizer.service.ts

```diff
@@ -154,48 +154,72 @@ export class SvgSanitizerService {
     return { tags: invalidTags, attrs: invalidAttrs };
   }
 
+  // SVG is considered an XMLDocument and DOMParser's parseFromtString method
+  // returns either a HTMLDocument or XMLDocument. Visit here for more info:
+  // https://developer.mozilla.org/en-US/docs/Web/API/DOMParser/parseFromString
+  // Document is the interface that both HTMLDocument and XMLDocument inherit
+  // properties from.
+  getSvgFromDataUri(dataURI: string): Document {
+    // Convert base64/URLEncoded data component to raw binary data
+    // held in a string.
+    let svgString = atob(dataURI.split(',')[1]);
+    let domParser = new DOMParser();
+    return domParser.parseFromString(svgString, 'image/svg+xml');
+  }
+
+  removeTagsAndAttributes(dataURI: string): string {
```

Comment:
This name is unclear because you're not passing in any tags or attributes. So it sounds like you are removing all tags and attributes which I suspect is not what you're trying to do.

Also it's weird that you're operating on a dataURI string here, because that string could be the dataURI for a PNG image, for example. So you might need to qualify that it is a data uri that represents an svg, or pass in the svg directly instead.

Reply:
Renamed the method and the parameter. Regarding the dataUri being that of a png or any other image format, a check has been done whenever this method has been called in the codebase. The check specifically inquires whether an image is an SVG before calling this method.